pkg/controller/replication_controller_test.go

```diff
@@ -844,3 +844,59 @@ func TestControllerUpdateStatusWithFailure(t *testing.T) {
 		t.Errorf("Expected 1 get and 2 updates, got %d gets %d updates", gets, updates)
 	}
 }
+
+func TestControllerBurstReplicas(t *testing.T) {
+	client := client.NewOrDie(&client.Config{Host: "", Version: testapi.Version()})
+	fakePodControl := FakePodControl{}
+	manager := NewReplicationManager(client)
+	manager.podControl = &fakePodControl
+
+	scaleFactor := 6
+	numReplicas := scaleFactor*BurstReplicas - 600
```

Comment:
The 600 is to make sure the manager doesn't barf at a non BurstReplica boundries. Scale factor can well be 2, I just want to make sure it auto resumes when it's seen BurstReplicas, and verify how/when it suspends.